platformio.ini

```diff
@@ -37,7 +37,7 @@ lib_extra_dirs =
   common/libs/
   common/test_libs/
 ; googletest requires pthread.
-build_flags = -Icommon/include/ -std=c++11 -DTEST_MODE -pthread
+build_flags = -Icommon/include/ -std=c++11 -DTEST_MODE -pthread -Wall -Werror
```

Comment:
-Wall -Werror is OK if everyone is using exactly the same compiler, so probably OK in the AVL case.  But in the native case, aren't we using whatever gcc you have on your host?

You could turn it on in CI if you wanted.

Reply:
We're using the host gcc, that's true, but it still feels like there's benefit in having Werror on native code (e.g. on our unit tests). This does have a risk that the warnings will be different for different people but it should still help write better code.

Reply:
What happens if my host GCC has a warning that yours doesn't?  Won't that mean the build is broken for me?